ext/ldap/ldap.c

```diff
@@ -4358,11 +4350,16 @@ PHP_FUNCTION(ldap_exop_passwd)
 		RETVAL_FALSE;
 	}
 
-	if (myargcount > 4) {
-		_php_ldap_controls_to_array(ld->link, lserverctrls, serverctrls, 0);
+cleanup:
+	if (lgenpasswd.bv_val != NULL) {
+		ldap_memfree(lgenpasswd.bv_val);
+	}
+	if (ldap_res != NULL) {
+		ldap_msgfree(ldap_res);
+	}
+	if (errmsg != NULL) {
+		ldap_memfree(errmsg);
 	}
```

Comment:
I think the if tests are redundant, all *free functions accepts NULL as an argument and do nothing, no?

Reply:
Right.

Reply:
Can I leave the if tests or shall I remove them for all the ``{ldap_mem,ber,ber_mem,e}free``? What about compatibility with older LDAP API? I see following pattern in several places in ``ldap.c``:

```
#if (LDAP_API_VERSION > 2000) || HAVE_ORALDAP || WINDOWS
      ldap_memfree(attribute);
#endif
```